daemon/src/maker_cfd.rs

```diff
@@ -632,141 +690,33 @@ where
-                },
-                taker_id,
-            )) => (proposal, *taker_id),
-            _ => {
-                anyhow::bail!("Order is in invalid state. Ignoring trying to accept the roll over request it.")
-            }
-        };
-
-        let dlc = cfd.open_dlc().context("CFD was in wrong state")?;
-
-        let oracle_event_id = oracle::next_announcement_after(
-            time::OffsetDateTime::now_utc() + cfd.order.settlement_interval,
-        )?;
-        let announcement = self
-            .oracle_actor
-            .send(oracle::GetAnnouncement(oracle_event_id))
-            .await?
-            .with_context(|| format!("Announcement {} not found", oracle_event_id))?;
+impl<O, M, T, W> Actor<O, M, T, W> {
+    async fn handle_reject_rollover(&mut self, msg: RejectRollOver) -> Result<()> {
+        tracing::info!(%msg.order_id, "Maker rejects a roll_over proposal" );
 
-        self.takers
-            .send(maker_inc_connections::TakerMessage {
-                taker_id,
-                msg: wire::MakerToTaker::ConfirmRollOver {
-                    order_id: proposal.order_id,
-                    oracle_event_id,
+        if !self
+            .rollover_actors
+            .send(
+                &msg.order_id,
+                maker_rollover::RejectRollOver {
+                    order_id: msg.order_id,
                 },
-            })
-            .await??;
-
-        let (sender, receiver) = mpsc::unbounded();
-        let contract_future = setup_contract::roll_over(
-            self.takers.clone().into_sink().with(move |msg| {
-                future::ok(maker_inc_connections::TakerMessage {
-                    taker_id,
-                    msg: wire::MakerToTaker::RollOverProtocol { order_id, msg },
-                })
-            }),
-            receiver,
-            (self.oracle_pk, announcement),
-            RolloverParams::new(
-                cfd.order.price,
-                cfd.quantity_usd,
-                cfd.order.leverage,
-                cfd.refund_timelock_in_blocks(),
-            ),
-            Role::Maker,
-            dlc,
-            self.n_payouts,
-        );
-
-        let this = ctx
-            .address()
-            .expect("actor to be able to give address to itself");
-
-        let task = async move {
-            let dlc = contract_future.await;
-
-            this.send(CfdRollOverCompleted { order_id, dlc })
-                .await
-                .expect("always connected to ourselves")
+            )
+            .await
+        {
+            tracing::warn!(%msg.order_id, "No active rollover");
         }
-        .spawn_with_handle();
 
-        self.roll_over_state = RollOverState::Active {
-            sender,
-            taker: taker_id,
-            _task: task,
-        };
+        self.projection_actor
+            .send(UpdateRollOverProposal {
+                order: msg.order_id,
+                proposal: None,
+            })
+            .await?;
```

Comment:
Move this into the rollover actor please.

Reply:
what about accept and completed? Should those also be handled in the rollover actor instead of the `maker_cfd::Actor`

Reply:
Yep, ideally the entire protocol is handled in the actor and the `maker_cfd::Actor` only gets informed about the outcome so it can update the state in DB.